portal-impl/src/com/liferay/portal/service/impl/CompanyLocalServiceImpl.java

```diff
@@ -220,9 +221,21 @@ public Company addCompany(
 			DBPartitionUtil.setDefaultCompanyId(company.getCompanyId());
 		}
 
+		com.liferay.portal.kernel.portlet.PortalPreferences
+			systemPortalPreferences = null;
+
```

Comment:
@tototrinh 
Why do you need to set this up before the use ? Why don't we move this to the part it's used?